blob/blob.go

```diff
@@ -30,19 +30,23 @@ const MaxShareVersion = 127
 // basic stateless checks over it.
 func New(ns namespace.Namespace, blob []byte, shareVersion uint8) *Blob {
 	return &Blob{
-		NamespaceId:      ns.ID,
+		NamespaceId:      ns.ID(),
 		Data:             blob,
 		ShareVersion:     uint32(shareVersion),
-		NamespaceVersion: uint32(ns.Version),
+		NamespaceVersion: uint32(ns.Version()),
 	}
 }
 
 // Namespace returns the namespace of the blob
-func (b *Blob) Namespace() namespace.Namespace {
-	return namespace.Namespace{
-		Version: uint8(b.NamespaceVersion),
-		ID:      b.NamespaceId,
-	}
+func (b *Blob) Namespace() (namespace.Namespace, error) {
+	return namespace.NewFromBytes(b.RawNamespace())
+}
```

Comment:
It seems we expect Blob to have valid namespace at that point? If so, we can avoid using exported constractor in favour of local(without validion) constractor. I will remove the need to return error here and:
- Method will be be much easier to use without `if err != nil ...`
- Won't need `(b *Blob) RawNamespace()` anymore

Reply:
Sure, in a follow up PR